software/controller/lib/core/nvparams.h

```diff
@@ -30,7 +35,13 @@ struct Structure {
   // Header info used to keep track of parameter info
   uint32_t crc{0};     // 32-bit CRC of remaining structure
   uint8_t count{0};    // Incremented on each write.
-  uint8_t version{0};  // Version of the structure.
+  uint8_t version{0};  // Version of the structure. If some fields in the structure change,
+                       // there is a risk of misinterpreting data that is in EEPROM during
+                       // initialization. This version is compared to the one in EEPROM and
+                       // the params in EEPROM are declared invalid if there is a mismatch.
+                       // Note that if the size of the structure changes, the chances that the crc
+                       // matches are non-existent and incrementing the version is not useful: the
+                       // EEPROM params will be invalid no matter what.
```

Comment:
What happens if there is a mismatch? Is the old data simply cleared and overwritten with default values?

Reply:
Yes. added in comment